qutip/tests/core/data/test_linalg.py

```diff
@@ -65,6 +71,13 @@ def test_mathematically_correct_Dense(self, method, opt):
                                    atol=1e-7, rtol=1e-7)
 
 
+    def test_singular(self):
+        A = qutip.num(2).data
+        b = qutip.basis(2, 1).data
+        with pytest.raises(ValueError):
+            test1 = _data.solve(A, b)
```

Comment:
We should check the error message here. `ValueError` is raised by many things and we should check that the correct `ValueError` is raised and that we didn't accidentally trigger some other error path.